[게시판 회원 기능] 티버 미션 제출합니다.

yjll1019 opened the pull request:
안녕하세요! 쏘쏘에게 리뷰를 받게 된 티버입니다! :)
이번 미션을 진행하면서 아직 수행하지 못한 부분이 있는데요.

- Inteceptor 를 이용해 로그인 기능을 처리하지 못한 점

- 세션에 대한 테스트 코드를 작성하지 못한 점

위의 두 부분을 아직 수행하지 못해 아쉽습니다. :(
이번 미션을 수행하면서 예상보다 시간이 오래 걸려서 필수 기능만 구현하고 pr을 보냅니다!
위의 두 기능은 필수 기능에 대해 조언을 얻은 후에 리팩토링하면서 차차 구현하겠습니다!

귀한 시간 내어 리뷰해주셔서 감사합니다!
많은 말씀 부탁드려요. ㅎㅎ

yjll1019 committed:
feat: add gradle dependency

yjll1019 committed:
feat: get/post practice

yjll1019 committed:
refactor: modify path

yjll1019 committed:
fix: add responseBody annotation

yjll1019 committed:
feat: click the button to move article-edit page

yjll1019 committed:
test: modify uri

yjll1019 committed:
feat: create and save article

yjll1019 committed:
feat: write the article

yjll1019 committed:
feat: look up the article

yjll1019 committed:
feat: look up the article by id

yjll1019 committed:
fix: find an article by id

yjll1019 committed:
feat: update the article

yjll1019 committed:
feat: delete the article

yjll1019 committed:
feat: move to index from article-edit

yjll1019 committed:
feat: setting devtools

yjll1019 committed:
fix : UI bug

yjll1019 committed:
refactor: remove duplicated html code

yjll1019 committed:
refactor: remove duplicated index logo

yjll1019 committed:
refactor: create ArticleService

yjll1019 committed:
refactor: divide package

yjll1019 committed:
refactor: divide package

yjll1019 committed:
refactor: modify return value

yjll1019 committed:
refactor: modify test cod

- fix bug
- extract duplicate fields

yjll1019 committed:
feat: update README.md

yjll1019 committed:
refactor: update README.md

yjll1019 committed:
refactor: remove MyblogApplication class

yjll1019 committed:
refactor: modify newline

yjll1019 committed:
refactor: modify return value of save method

yjll1019 committed:
refactor: add ArticleService field

- remove ArticleRepository field

yjll1019 committed:
refactor: for convention

yjll1019 committed:
refactor: create NotFoundArticleException

yjll1019 committed:
refactor: modify field name

yjll1019 committed:
refactor: modify newline

yjll1019 committed:
refactor: extract duplicate code

yjll1019 committed:
refactor: create matchId method

yjll1019 committed:
refactor: modify return value of findAll method

yjll1019 committed:
refactor: modify annotation

yjll1019 committed:
refactor: create update method in Article class

yjll1019 committed:
refactor: modify method name, location

yjll1019 committed:
test: create ArticleRepositoryTests

yjll1019 committed:
refactor: modify field type of id

yjll1019 committed:
refactor: remove setter of title, contents, coverUrl field

yjll1019 committed:
refactor: modify semicolon position

yjll1019 committed:
refactor: modify URL

yjll1019 committed:
test: modify findAll method

yjll1019 committed:
test: modify uri

yjll1019 committed:
test: add ability to check location of header

yjll1019 committed:
refactor: remove ArticleService class

yjll1019 committed:
refactor: modify path

yjll1019 committed:
refactor: modify id field type

yjll1019 committed:
refactor: modify injection

yjll1019 committed:
refactor: fix bug

yjll1019 committed:
refactor: for code convention

yjll1019 committed:
refactor: modify properties

yjll1019 committed:
refactor: add spring-data-jap and h2

- create ArticleDto

yjll1019 committed:
feat: mysql 연동

yjll1019 committed:
docs: 기능 목록 추가

yjll1019 committed:
feat: MySQL 사용 설정

yjll1019 committed:
fix: 템플릿 파일명 중복제거 및 오타 수정

yjll1019 committed:
feat: signup get 구현 및 테스트 추가

yjll1019 committed:
add: User 및 UserDto 클래스 추가

yjll1019 committed:
feat: users post 데이터 받아오는 기능 구현

yjll1019 committed:
feat: 이름 및 패스워드 규칙 추가

이름은 최소 2글자, 최대 10글자까지 입력 가능.
패스워드는 최소 8글자 이상으로 알파벳 소문자, 대문자, 숫자, 특수문자의 조합이다.

yjll1019 committed:
feat: 프론트엔드에 입력 검증 기능 추가

yjll1019 committed:
feat: 비밀번호 확인 기능 동작 추가

yjll1019 committed:
feat: 회원가입 실패 시 에러 메시지 페이지에 노출

yjll1019 committed:
feat: 동일한 email 회원가입 불가 기능 구현

yjll1019 committed:
test: 회원등록 기능 테스트 구현

yjll1019 committed:
feat: 회원조회 기능 구현

yjll1019 committed:
test: 회원조회 기능 테스트 구현

yjll1019 committed:
feat: 로그인 기능 구현

- 로그인 성공 시 메인 화면을 띄우고 우측 상단에 사용자 이름을 띄운다.
- 로그인 실패 시 상황에 맞는 실패 메시지를 띄운다.

yjll1019 committed:
fix: 세션 생성 방법 수정

yjll1019 committed:
test: 로그인 기능 테스트 구현

yjll1019 committed:
feat: 로그아웃 기능 구현

yjll1019 committed:
test: 로그아웃 기능 테스트 구현

yjll1019 committed:
refactor: 로그인 시 세션 관리 수정

yjll1019 committed:
test: 로그인 기능 테스트 수정

yjll1019 committed:
feat: 로그인 한 유저가 로그인/회원가입 화면에 접근할 경우 메인 화면으로 이동

yjll1019 committed:
refactor: 로그아웃 시 세션 처리 방법 수정

yjll1019 committed:
refactor: 컨트롤러명 수정

yjll1019 committed:
test: UserControllerTest 테스트 코드 수정

yjll1019 committed:
feat: mypage, mypage-edit 접근 권한 기능 추가

- session확인 시 email로 확인하도록 수정

yjll1019 committed:
test: 로그인하지 않았을 경우 mypage, mypage-edit 이동 실패 테스트 구현

- 쿠키에서 session 확인하도록 LoginControllerTests 메소드 수정

yjll1019 committed:
feat: 회원 탈퇴 기능 구현

yjll1019 committed:
refactor: 회원 탈퇴 시 DeleteMapping으로 수정

yjll1019 committed:
feat: 회원 수정 기능 구현

yjll1019 committed:
docs: 기능 목록 수정

yjll1019 committed:
refactor: dto.request 패키지 생성

yjll1019 committed:
test: test 코드 중복 제거

- LogoutControllerTests 삭제
- session확인 코드 삭제

yjll1019 committed:
refactor: 코드 컨벤션을 위한 수정

yjll1019 committed:
Merge pull request #3 from yjll1019/yjll1019_level_2

Yjll1019 level 2

yjll1019 committed:
refactor: properties 설정 파일 수정

yjll1019 committed:
feat: siguup 페이지에서 login 페이지로 이동하도록 url 수정

yjll1019 committed:
Merge branch 'yjll1019' into yjll1019

ssosso reviewed (changes requested):
안녕하세요, 티버~ 쏘쏘입니다:)
말씀주신 내용 감안하고 피드백하였습니다~ interceptor는 advanced 영역이기 때문에 세션에 대한 테스트가 먼저 구현되었으면 합니다. 세션에 대한 테스트는 어떻게 할지 감이 오지 않는다면 다른 크루의 코드를 참고해보시면 어떨까요~?

yjll1019 committed:
fix: git conflict 해결 과정 중 발생한 오류 수정

yjll1019 committed:
refactor: ArticleDto.valueOfArticle 메소드 구현

yjll1019 committed:
test: ArticleDto.valueOfArticle 메소드 테스트 구현

yjll1019 committed:
refactor: 존재하지 않는 Article 요청 시 error 메세지 반환

yjll1019 committed:
refactor: valueOfArticle 메소드 오버로딩

yjll1019 committed:
refactor: 변수명 변경

yjll1019 committed:
fix: ArticleDto 관련 버그 수정

yjll1019 committed:
refactor: confirmSession 메소드 생성

yjll1019 committed:
test: User의 패스워드가 일치하지 않는 경우 테스트 코드 구현

yjll1019 committed:
test: 회원가입 실패 시 테스트 케이스 추가

yjll1019 committed:
fix: LoginController의 메소드 반환 값 수정

yjll1019 committed:
test: 테스트 코드 중복 제거

yjll1019 committed:
refactor: 변수명 수정

yjll1019 committed:
refactor: User 생성자 수정

yjll1019 committed:
refactor: LoginService 클래스 분리

- Login, NotFoundUserEmail, NotFoundMatchPassword 예외 클래스 구현
- GlobalExceptionHandler 클래스 구현

yjll1019 committed:
test: Login 실패 시 테스트 코드 수정

yjll1019 committed:
refactor: 존재하지 않는 게시글에 대한 예외 처리 수정

yjll1019 committed:
test: 존재하지 않는 게시글에 대한 테스트 코드 구현

yjll1019 committed:
refactor: service클래스 분리

yjll1019 committed:
feat: BasicAuthInterceptor, WebMvcConfig 클래스 구현

yjll1019 committed:
test: 로그인 되어있는 경우 로그인 실패하는 테스트 구현

yjll1019 committed:
feat: Interceptor 구현

yjll1019 committed:
test: Interceptor 테스트 코드 구현

yjll1019 committed:
refactor: 메소드명 변경, 반환 값 수정

yjll1019 committed:
refactor: 메소드명 변경, 반환 값 수정

yjll1019 committed:
refactor: 코드 컨벤션을 위한 리팩토링

yjll1019 committed:
refactor: Interceptor 와 config 수정

yjll1019 committed:
refactor: confirmSession 수정

yjll1019 committed:
refactor: session 관련 테스트 코드 구현

yjll1019 committed:
refactor: 코드 컨벤션을 위한 리팩토링

yjll1019 commented:
안녕하세요 쏘쏘~! 
구현하면서 여쭤보고 싶은 부분이 있는데요!

- 현재 BindingResult객체에 에러가있을 때 처리하는 부분을 컨트롤러 클래스에 두었는데 이 부분을 서비스 클래스에서 처리해주어도 괜찮을까요?? 

- 인터셉터를 구현한 경우 제가 컨트롤러에서 구현한 세션을 확인하는 작업과 사용자를 구별하는 기능이 중복되는거 같은데 인터셉터에서도 필요에 따라 로그인한 사용자를 구별하고 컨트롤러에서도 구별해야할까요??

yjll1019 committed:
refactor: GlobalExceptionHandler 클래스 이동

ssosso reviewed (approved):
티버~ 적용해주신 부분 확인했어요 👍  좋습니다:)
질문주신 내용은 아래와 같이 답변드립니다~

- 현재 BindingResult객체에 에러가있을 때 처리하는 부분을 컨트롤러 클래스에 두었는데 이 부분을 서비스 클래스에서 처리해주어도 괜찮을까요??
  - BindingResult를 처리하는 별도의 Service를 만든다면 가능할 것 같아요~ 기존에 있는 Service들로 옮겨가면 어색할 것 같네요:) Controller의 private 메서드로 있어도 되고, 재활용성을 높이고 싶다면 별도의 클래스를 만드는 것이 좋겠네요~

- 인터셉터를 구현한 경우 제가 컨트롤러에서 구현한 세션을 확인하는 작업과 사용자를 구별하는 기능이 중복되는거 같은데 인터셉터에서도 필요에 따라 로그인한 사용자를 구별하고 컨트롤러에서도 구별해야할까요??
  - 인터셉터에서만 진행하시면 됩니다:) 인터셉터의 구조 자체가 컨트롤러 로직에 존재하는 중복 전/후처리를 위한 것이기 때문입니다~

이번 단계는 이만 마무리하도록 할게요:)
코멘트 드린 내용에 대해 작업하시는 부분은 다음 리뷰어에게 새로운 의견이 있는지 물어봐주시는 것도 좋을 것 같아요~
고생하셨습니다~
